File: Front/ComparaItens-App_/src/app/models/Product.ts
import { Category } from './Category';
import { Manufacturer } from './Manufacturer';
import { SpecificationItem } from './SpecificationItem';

export interface Product {
  id: number;
  description: string;
  manufecturerId: number;
  model: string;
  categoryId: number;
  yearOfManufacture: number;
  image: string;
  folder: string;
  manufacturer: Manufacturer;
  category: Category;
  specificationItem: SpecificationItem[];
}

File: Front/BlazorAppBuscaItens/BlazorAppBuscaItens/Model/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace BlazorAppBuscaItens.Model
{
    public class Product
    {
        public int Id { get; set; }

        public string Description { get; set; }

        public int ManufecturerId { get; set; }

        public string Model { get; set; }

        public int CategoryId { get; set; }

        public int YearOfManufacture { get; set; }
        public string Image { get; set; }
        public string Folder { get; set; }

        public IList<ProductItem> ProductItems { get; set; }

        public Manufacturer Manufecturer { get; set; }

        public Category Category { get; set; }
    }
}
